core/pkg/artifacts/file_cache.go

```diff
@@ -0,0 +1,147 @@
+package artifacts
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"io"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	atomicfile "github.com/natefinch/atomic"
+	"github.com/wandb/wandb/core/pkg/utils"
+)
+
+type FileCache struct {
+	root    string
+	enabled bool
+}
+
+func NewFileCache(enabled bool) FileCache {
```

Comment:
are there cases where you want this to be controllable? I'm thinking about some hacky cases where Ishita used tricks with the cache dir path for skip cache? Maybe we never need to 🤷

Reply:
final note, one previous engineer told me boolean variables in constructor should never be allowed :P I am reminded of why when I read this PR: https://github.com/wandb/wandb/pull/7366/files#diff-64bffd2953311d041b82ef9c6bae578f687e5cc2ef825181a021e2fb7843c3cbR43 consider making it a little easier for people to parse :)

Reply:
I'm pretty sure that what I have now works for what you want
1. we explicitly pass in a path, which we get from a separate `UserCacheDir` function
2. we don't have an enable/disable, we now have a HashOnlyCache